helpers/lib_tasks_pytest.py

```diff
@@ -874,6 +874,121 @@ def run_coverage_report(  # type: ignore
         _publish_html_coverage_report_on_s3(aws_profile)
 
 
+def _get_inclusion_settings(target_dir: str) -> Tuple[str, Optional[str]]:
+    """
+    Determine include/omit glob patterns for the coverage report.
+
+    :param target_dir: directory for coverage stats; use "." to indicate all directories
+    :return: tuple (include_in_report, exclude_from_report) where
+        - include_in_report is the glob pattern to include
+        - exclude_from_report is a comma‑separated glob pattern to omit, or ``None``
+    """
+    if target_dir == ".":
+        include_in_report: str = "*"
+        exclude_from_report: Optional[str] = None
+        if hserver.skip_submodules_test():
+            submodule_paths: List[str] = hgit.get_submodule_paths()
+            exclude_from_report = ",".join(
+                f"{path}/*" for path in submodule_paths
+            )
+    else:
+        include_in_report = f"*/{target_dir}/*"
+        exclude_from_report = None
+    return include_in_report, exclude_from_report
+
+
+def _run_coverage(
+    ctx,
+    suite: str,
+    target_dir: str,
+    generate_html_report: bool,
+) -> str:
+    """
+    Internal helper to run coverage for a given suite (fast/slow) and emit XML.
+
+    :param suite: one of "fast" or "slow"
+    :param target_dir: coverage target directory
+    :param generate_html_report: whether to produce HTML output
+    :return: filename of the suite‑specific coverage data file
```

Comment:
Can use provide use-cases for this return value? 
If we don't use it, let's remove it everywhere.